pyquil/api/_compiler.py

```diff
@@ -191,3 +206,41 @@ def __init__(
 
     def native_quil_to_executable(self, nq_program: Program, **kwargs: Any) -> QuantumExecutable:
         return nq_program
+
+
+class IncompatibleBackendForQuantumProcessorIDWarning(Warning):
+    pass
+
+
+def validate_backend_for_quantum_processor_id(
+    quantum_processor_id: str, backend: Optional[TranslationBackend]
+) -> TranslationBackend:
```

Comment:
Naming: this does more than validate, it selects or coerces. `validate` intuitively returns a `bool` or throws an exception